internal/provider/roles/client.go

```diff
@@ -0,0 +1,195 @@
+package roles
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/permitio/permit-golang/pkg/models"
+	"github.com/permitio/permit-golang/pkg/permit"
+)
+
+type RoleClient struct {
+	client *permit.Client
+}
+
+type RoleMethods interface {
+	RoleRead(ctx context.Context, data RoleModel) (RoleModel, error)
+	RoleCreate(ctx context.Context, actions map[string]models.ActionBlockEditable, resourcePlan *RoleModel) error
+	RoleUpdate(ctx context.Context, resourcePlan *RoleModel) error
+}
+
+func (d *RoleClient) RoleRead(ctx context.Context, data RoleModel) (RoleModel, error, diag.Diagnostics) {
```

Comment:
change this to return pointer, don't return diag.Diagnostics, wrap this in role_resource and return an error instead